src/controller/CHIPDevice.cpp

```diff
@@ -636,89 +629,11 @@ CHIP_ERROR Device::EstablishConnectivity(Callback::Callback<OnDeviceConnected> *
     return CHIP_NO_ERROR;
 }
 
-void Device::AddResponseHandler(uint8_t seqNum, Callback::Cancelable * onSuccessCallback, Callback::Cancelable * onFailureCallback,
-                                app::TLVDataFilter tlvDataFilter)
-{
-    mCallbacksMgr.AddResponseCallback(mDeviceId, seqNum, onSuccessCallback, onFailureCallback, tlvDataFilter);
-}
-
-void Device::CancelResponseHandler(uint8_t seqNum)
-{
-    mCallbacksMgr.CancelResponseCallback(mDeviceId, seqNum);
-}
-
-void Device::AddIMResponseHandler(app::CommandSender * commandObj, Callback::Cancelable * onSuccessCallback,
-                                  Callback::Cancelable * onFailureCallback)
-{
-    // We are using the pointer to command sender object as the identifier of command transactions. This makes sense as long as
-    // there are only one active command transaction on one command sender object. This is a bit tricky, we try to assume that
-    // chip::NodeId is uint64_t so the pointer can be used as a NodeId for CallbackMgr.
-    static_assert(std::is_same<chip::NodeId, uint64_t>::value, "chip::NodeId is not uint64_t");
-    chip::NodeId transactionId = reinterpret_cast<chip::NodeId>(commandObj);
-    mCallbacksMgr.AddResponseCallback(transactionId, 0 /* seqNum, always 0 for IM before #6559 */, onSuccessCallback,
-                                      onFailureCallback);
-}
-
-void Device::CancelIMResponseHandler(app::CommandSender * commandObj)
-{
-    // We are using the pointer to command sender object as the identifier of command transactions. This makes sense as long as
-    // there are only one active command transaction on one command sender object. This is a bit tricky, we try to assume that
-    // chip::NodeId is uint64_t so the pointer can be used as a NodeId for CallbackMgr.
-    static_assert(std::is_same<chip::NodeId, uint64_t>::value, "chip::NodeId is not uint64_t");
-    chip::NodeId transactionId = reinterpret_cast<chip::NodeId>(commandObj);
-    mCallbacksMgr.CancelResponseCallback(transactionId, 0 /* seqNum, always 0 for IM before #6559 */);
-}
-
-void Device::AddReportHandler(EndpointId endpoint, ClusterId cluster, AttributeId attribute,
-                              Callback::Cancelable * onReportCallback)
-{
-    mCallbacksMgr.AddReportCallback(mDeviceId, endpoint, cluster, attribute, onReportCallback);
-}
-
-CHIP_ERROR Device::SendReadAttributeRequest(app::AttributePathParams aPath, Callback::Cancelable * onSuccessCallback,
-                                            Callback::Cancelable * onFailureCallback, app::TLVDataFilter aTlvDataFilter)
+CHIP_ERROR Device::PreparePeer()
 {
     bool loadedSecureSession = false;
-    uint8_t seqNum           = GetNextSequenceNumber();
-    aPath.mNodeId            = GetDeviceId();
-
     ReturnErrorOnFailure(LoadSecureSessionParametersIfNeeded(loadedSecureSession));
```

Comment:
I see that `PreparePeer()` calls `LoadSecureSessionParametersIfNeeded()`: does this require that a secure session was already established at some point by the device that is the target of the connection here?

For example, would the following be valid with your change?
1. device A is commissioned by device B (device A establishes SecureSession to B)
2. commissioning SecureSession is closed
3. device B establishes new SecureSession to A using your changes

Reply:
Yes, a new `InteractionPeerDelegate` implementation can be added where a CASE session is setup with the provisioned operational credential.